src/ng/compile.js

```diff
@@ -983,7 +983,7 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
   var bindingCache = createMap();
 
   function parseIsolateBindings(scope, directiveName, isController) {
-    var LOCAL_REGEXP = /^\s*([@&<]|=(\*?))(\??)\s*(\w*)\s*$/;
+    var LOCAL_REGEXP = /^\s*([@&<]|=(\*?))(\??)\s*([$]?\w*[$]?)\s*$/;
```

Comment:
I don't think there is any harm in allowing `$` anywhere in the name, so I would change `[$]?\w*[$]?` to `[\w$]*`.